eng/build.ps1

```diff
@@ -15,6 +15,40 @@
 # `.\build.ps1 -testDesktop`.
 
 [CmdletBinding(PositionalBinding=$false)]
+[Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSAvoidAssignmentToAutomaticVariable', '', Scope="Function", Target="TestUsingRunTests")]
+[Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSAvoidDefaultValueSwitchParameter', '')]
+[Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSAvoidGlobalVars', '')]
+[Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSReviewUnusedParameter', 'binaryLog')]
+[Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSReviewUnusedParameter', 'bootstrapConfiguration')]
+[Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSReviewUnusedParameter', 'buildServerLog')]
+[Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSReviewUnusedParameter', 'collectDumps')]
+[Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSReviewUnusedParameter', 'configuration')]
```

Comment:
the "scope" they are declared in is just the try-catch which they are not used in. They are used in child scopes (i.e. other functions) implicitly. to remove this we would need to explicitly pass these to all the functions that consume them instead of assuming that they are defined in the parent scope.

Reply:
for example:

[`$configuration`](https://github.com/dotnet/roslyn/blob/4cdfdba8ba641ca8d204d3faacf010af91544064/eng/build.ps1#L19) is only read from implicitly in [`BuildSolution()`](https://github.com/dotnet/roslyn/blob/4cdfdba8ba641ca8d204d3faacf010af91544064/eng/build.ps1#L246) and [`TestUsingRunTests()`](TestUsingRunTests()). We could instead explicitly pass it as an argument to both those functions if we wanted to get rid of the warning.